static/js/popovers.js

```diff
@@ -143,7 +143,7 @@ function get_custom_profile_field_data(user, field, field_types, dateFormat) {
 
     switch (field_type) {
         case field_types.DATE.id:
-            profile_field.value = moment(field_value.value).format(dateFormat);
+            profile_field.value = dateFormat.format(new Date(field_value.value));
```

Comment:
I think this needs to be `parseISO`. Otherwise this subtracts a day for users in the western hemisphere.

```
> {parseISO} = require("date-fns")
> dateFormat = new Intl.DateTimeFormat("default", {dateStyle: "long"})
> dateFormat.format(new Date('2000-01-01'))
'December 31, 1999'
> dateFormat.format(parseISO('2000-01-01'))
'January 1, 2000'
```

Reply:
There seems to be regex <a href = 'https://github.com/date-fns/date-fns/issues/1965'>issue<a/> in <code>date-fns</code> which throws an <code>Invalid-Date</code> if a date like '1900-1-1' is passed within <code>parseISO</code> . 
Not sure if something like <code>dateFormat.format(parseISO(formatISO(new Date(field_value.value))))</code> could be a 'hacky' approach to resolve this .

Reply:
Oh I see, `populate_db.py` and `openapi/zulip.yaml` use “1900-1-1” and “2000-1-1” as example values. These really are invalid ISO 8601 dates that should be written “1900-01-01” and “2000-01-01”, so I don’t blame `parseISO` for rejecting them.

Let’s fix these invalid dates and fix the validator to reject them: #16903.

Reply:
Merged #16903, so you should be able to rebase this on top of that.

Reply:
Cool ! Made the necessary changes :+1:  , Will move on to fix failing node-tests if changes look fine .